### PR TITLE
Mock event projection

### PR DESCRIPTION
**Pull request by condron**
MockStreamStore Updates
adds support for event type 'et-' and category 'ce-` standard projections in Mock Eventstore used for netcore tests
adds support for 'All' subscription 
corrects implementations of read forward and read backward 
corrects subscriptions 
enables and corrects tests for confirming matching behavior between ES and Mock

### CODE REVIEW

**Comment by joshkempner** [comment]
Can you merge from master back into this branch? It looks like this pull request also includes changes from the recently merged #33, which is confusing.

**Comment by condron** [comment]
Hi Josh, 
I pulled the changes in from the later service branch before the pull request so the commits to master are listed here as well. I've tried to re-merge from master but I just get an already up to date message. 

The only relevant files here are the one in the ReactiveDomain\Testing folder. If you know how to clear those commits out, let me know.
Chris

File: src/ReactiveDomain.Testing/EventStore/MockStreamStoreConnection.cs [hunk 1]
```diff
@@ -62,205 +52,341 @@ public void Close()
...
+            return subscription;
+
         }
 
-        public void DeleteStream(string stream, int expectedVersion, UserCredentials credentials = null)
-        {
-            throw new NotImplementedException();
+        public void DeleteStream(string stream, int expectedVersion, UserCredentials credentials = null) {
+            if (stream.StartsWith(CategoryStreamNamePrefix, StringComparison.OrdinalIgnoreCase) ||
+               stream.StartsWith(EventTypeStreamNamePrefix, StringComparison.OrdinalIgnoreCase)) {
```
[the remark this answers is not in the document]
Comment (condron) [reply src/ReactiveDomain.Testing/EventStore/MockStreamStoreConnection.cs]: fixed in the next changes, with tests even.

File: src/ReactiveDomain.Testing/EventStore/StreamStoreReadTests.cs [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+                    count);
+
+                Assert.True(count == slice.Events.Length, "Failed to read events forward");
+                Assert.Equal(startFrom, slice.FromEventNumber);
+                Assert.Equal(_lastEvent, slice.LastEventNumber);
+                Assert.Equal(startFrom + count, slice.NextEventNumber);
+                Assert.False(slice.IsEndOfStream);
+                Assert.Equal(ReadDirection.Forward, slice.ReadDirection);
+                Assert.True(string.CompareOrdinal(_streamName, slice.Stream) == 0);
+                j = 3;
```
[the remark this answers is not in the document]
Comment (condron) [reply src/ReactiveDomain.Testing/EventStore/StreamStoreReadTests.cs]: switch over halfway through, I'll go back and clean up

File: src/ReactiveDomain.Testing/EventStore/StreamStoreSubscriptionTests.cs [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+
+                AssertEx.IsOrBecomesTrue(() => Interlocked.Read(ref evtCount) == 1);
+                AssertEx.IsOrBecomesTrue(() => Interlocked.Read(ref evtNumber) == 11);
+                AppendEvents(5, conn, _streamName, 12);
+
+                AssertEx.IsOrBecomesTrue(() => Interlocked.Read(ref evtCount) == 6);
+                AssertEx.IsOrBecomesTrue(() => Interlocked.Read(ref evtNumber) == 16);
+
+                sub.Dispose();
+                AssertEx.IsOrBecomesTrue(() => Interlocked.Read(ref dropped) == True, msg: "Failed to handle drop");
```
[the remark this answers is not in the document]
Comment (condron) [reply src/ReactiveDomain.Testing/EventStore/StreamStoreSubscriptionTests.cs]: I like knowing nothing is hanging around

**Comment by condron** [comment]
This should be it for this PR

**Pull request merged**